File: src/dataset/go/secret_handshake_Exercism_go_rootulp/openAI/ChatGPT4_secret_handshake_v1.go
package secret

var operations = []string{"wink", "double blink", "close your eyes", "jump"}

func Handshake(code uint) (handshake []string) {
	for i := 0; i < len(operations); i++ {
		if code&(1<<i) != 0 {
			handshake = append(handshake, operations[i])
		}
	}
	if code&(1<<len(operations)) != 0 {
		reverseSliceInPlace(handshake)
	}
	return handshake
}

func reverseSliceInPlace(input []string) {
	for i, j := 0, len(input)-1; i < j; i, j = i+1, j-1 {
		input[i], input[j] = input[j], input[i]
	}
}

File: src/dataset/javascript/secret-handshake_exercism-javascript-ffflorian/openAI/ChatGPT4_secret-handshake_v1.js
const availableCommands = ['wink', 'double blink', 'close your eyes', 'jump'];

export function commands(input) {
  const result = [];
  for (let i = 0; i < 4; i++) {
    if (input & (1 << i)) {
      result.push(availableCommands[i]);
    }
  }
  return (input & 16) ? result.reverse() : result;
}